core/domain/exp_domain.py

```diff
@@ -2213,6 +2246,167 @@ def _convert_states_v28_dict_to_v29_dict(cls, states_dict):
+
+            # Adding image_id, getting image info and removing filepath from
+            # image tag present in hint interaction.
+            for hint_index, hint in enumerate(
+                    state_dict['interaction']['hints']):
+                # Adding image.
+                hint_html = hint['hint_content']['html']
+                (new_hint_html, image_counter) = (
+                    html_validation_service.add_image_id_in_image_tag(
+                        hint_html, image_counter))
+
+                image_info_list = image_info_list + (
+                    html_validation_service.get_image_info_from_html(
+                        new_hint_html))
+
+                state_dict['interaction']['hints'][hint_index][
+                    'hint_content']['html'] = (
+                        html_validation_service.
+                        remove_filepath_from_image_tag(new_hint_html))
+
+            # Adding image_id, getting image info and removing filepath from
+            # image tag present in solution interaction.
+            if state_dict['interaction']['solution']:
+                solution_html = state_dict[
+                    'interaction']['solution']['explanation']['html']
+                (new_solution_html, image_counter) = (
+                    html_validation_service.add_image_id_in_image_tag(
+                        solution_html, image_counter))
+
+                image_info_list = image_info_list + (
+                    html_validation_service.get_image_info_from_html(
+                        new_solution_html))
+
+                state_dict['interaction']['solution']['explanation']['html'] = (
+                    html_validation_service.remove_filepath_from_image_tag(
+                        new_solution_html))
+
+            # Adding image_id, getting image info and removing filepath from
+            # image tag present in following given below interactions.
+            if state_dict['interaction']['id'] in (
+                    'MultipleChoiceInput', 'DragAndDropSortInput'):
+                for value_index, value in enumerate(
+                        state_dict['interaction']['customization_args'][
+                            'choices']['value']):
+                    (new_value, image_counter) = (
+                        html_validation_service.add_image_id_in_image_tag(
+                            value, image_counter))
+
+                    image_info_list = image_info_list + (
+                        html_validation_service.get_image_info_from_html(
+                            new_value))
+
+                    state_dict['interaction']['customization_args'][
+                        'choices']['value'][value_index] = (
+                            html_validation_service.
+                            remove_filepath_from_image_tag(new_value))
+
+            # Add image info in image assets.
+            for image in image_info_list:
+                image_id = unicode(image['id'])
```

Comment:
Why are we still doing this?

Reply:
We are expecting the image_id to be a string (image-1) right?

Reply:
yeah, but one thing imag_id will be ```image_id_1```, not ```image-1```.
Is that fine?

Reply:
Fine with either but I would prefer image_1 (length of the string is comparatively less, easy to extract the int part). Whatever you chose make sure to document and validate this in imageAssets class (validate in the validation method).